### ISSUE TITLE
Semantic analysis error not reported within lambda

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#4098]
This doesn't generate an error, though it's clearly invalid:

```py
lambda: xyz
```

This generates an error:

```py
lambda: 1 + ''
```

It looks like type checking errors are reported within a lambda but semantic analysis errors are lost. 

This is not a regression -- this bug has been around for a long time.
